quickjs-libc.c

```diff
@@ -3846,17 +3846,45 @@ static JSValue js_print(JSContext *ctx, JSValue this_val,
     const char *str;
     size_t len;
 
+#ifdef _WIN32
+    wchar_t *wstr;
+    DWORD written;
+    HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+    if (hConsole == INVALID_HANDLE_VALUE)
+        return JS_EXCEPTION;
+#endif
+
     for(i = 0; i < argc; i++) {
         if (i != 0)
+#ifdef _WIN32
+            WriteConsoleW(hConsole, L" ", 1, &written, NULL);
+#else
             putchar(' ');
+#endif
         str = JS_ToCStringLen(ctx, &len, argv[i]);
         if (!str)
             return JS_EXCEPTION;
+#ifdef _WIN32
+        // Calculate the size of the wide string
+        int wlen = MultiByteToWideChar(CP_UTF8, 0, str, len, NULL, 0);
+        wstr = (wchar_t*)malloc((wlen + 1) * sizeof(wchar_t));
+        // Convert UTF-8 string to wide string
+        MultiByteToWideChar(CP_UTF8, 0, str, len, wstr, wlen);
+        wstr[wlen] = L'\0'; // Null-terminate the wide string
+        // Output the wide string to the console
+        WriteConsoleW(hConsole, wstr, wlen, &written, NULL);
+        free(wstr);
```

Comment:
Please do not use `malloc`. Use `js_malloc` instead and check for potential allocation failure.